perses/annihilation/lambda_protocol.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import print_function
+import numpy as np
+import copy
+import logging
+import traceback
+from openmmtools.alchemy import AlchemicalState
+
+# make something hyperbolic or something to go from on to off to on
+default_hybrid_functions = {
```

Comment:
We should probably document what these three sets of functions are and make sure they describe consistent protocols.

The first set (`default_hybrid_functions`) are clearly for Lepton, but why do we have both `python_hybrid_functions` and `python_reverse_functions`, what differentiates them, and where are they used?

Reply:
@jchodera We have both `python_hybrid_functions` and `python_reverse_functions` because the approach that sets the context parameters manually (with Python functions) still runs `lambda` from 0 to 1 on every protocol. I don't think there is a way to easily auto-generate "reverse" functions from the Python expressions (unlike the Lepton strings) without being super hacky/engaging in excessive metaprogramming.